sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -202,7 +208,9 @@ def run_with_workflow(self, sale_workflow):
             )
 
         if sale_workflow.register_payment:
-            self._register_payments(
+            self.with_context(
+                property_payment_journal_id=sale_workflow.property_payment_journal_id
```

Comment:
Is it perhaps the case that at this point, the process is still company agnostic, and you should fetch the value of this company dependent field at a later stage (with the order's company if not already set)?

Reply:
@kevinkhao